rocks-lib/src/project/rocks.rs

```diff
@@ -0,0 +1,79 @@
+//! Structs and utilities for `rocks.toml`
+
+use std::collections::HashMap;
+
+use serde::Deserialize;
+
+use crate::{
+    package::{PackageName, PackageReq, PackageVersion},
+    rockspec::{
+        ExternalDependencySpec, PerPlatform, PlatformSupport, RockDescription, RockSource, Rocks,
+    },
+};
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct RocksToml {
+    package: PackageName,
+    version: PackageVersion,
+    description: RockDescription,
+    supported_platforms: PlatformSupport,
+    dependencies: PerPlatform<Vec<PackageReq>>,
+    build_dependencies: PerPlatform<Vec<PackageReq>>,
+    external_dependencies: PerPlatform<HashMap<String, ExternalDependencySpec>>,
+    test_dependencies: PerPlatform<Vec<PackageReq>>,
+    //source: PerPlatform<RockSource>,
+    //pub build: BuildSpec,
+    //pub test: TestSpec,
```

Comment:
@mrcjkb I'm a little lost on how to properly implement `Deserialize` for `RockSource` and the others without everything erroring. Is trying to make `RockSource` implement `Deserialize` flawed? Should some other work be done on `PerPlatform`/`PerPlatformWrapper` instead?

Reply:
You'll probably want to do something similar to the `impl<T> FromLua for PerPlatform<T>`.